src/main.c

```diff
@@ -738,6 +756,21 @@ int main(int argc, char *argv[])
 		goto exit;
 	}
 
+	if(cfg.saml_port != 0) {
+		// Wait for the SAML token from the HTTP GET request
+		if (start_http_server(&cfg) != 0) {
```

Comment:
Given that this is a blocking call that is expected to return a session id, I suggest this function be renamed. To me, `start_http_server` implies that there will be a long-running background thread.